### DISCUSSION TITLE
Curve subdivision based on its tangent

### DISCUSSION THREAD

**Discussion by Durman** [JacquesLucke/animation_nodes#1845] [Q&A]
I have next animation which a managed to achieve with a subprogram in Sverchok. I'm trying simulate vehicle movement where movement on some parts of a curve is more expensive and it seems is the perfect task for Animation Nodes. The question is whether it's possible to do without subprograms in AN?
![running point](https://user-images.githubusercontent.com/28003269/163313427-ff57d8ef-32e4-4362-9b10-ca43fc66d072.gif)
![2022-04-13_21-17-15](https://user-images.githubusercontent.com/28003269/163313451-e277bb9e-274a-4d95-bb7f-e671e8434b1a.png)

**Comment by OmarEmaraDev** [comment]
I don't think it can be done without a loop at the moment. The best I can think of without a loop would be to construct a new curve from the original one with masked points based on the evaluated curvature and evaluate that new curve using the resolution method.

**Reply by Durman** [reply]
Thanks. I tried some sort of this variant in GN but did not manage to achieve satisfactory result. Also this approach might be quite less efficient sometimes I guess. It seems that this task is more for simulation than for modeling and animating.
